File: angular-client/src/app/form/validators/email-rule.ts
import { AbstractControl, ValidationErrors } from '@angular/forms';
import { AbstractRule } from './abstract-rule';
import { ValidationResultFactory } from './validation-result-factory';

export class EmailRule extends AbstractRule {
  validate(control: AbstractControl): ValidationErrors | null {
    const resultFactory = new ValidationResultFactory(
      'EmailRule',
      this.validationService
    );

    if (control.value) {
      if (control.value.indexOf('@') < 0) {
        resultFactory.addResult('InvalidEmail');
      }
    }

    return resultFactory.validationErrors;
  }
}

File: couch-server/src/main/java/com/fronde/server/services/validation/rule/EmailRule.java
package com.fronde.server.services.validation.rule;

import com.fronde.server.services.validation.ValidationResultFactory;

public class EmailRule extends AbstractRule {

  @Override
  public void validate(Object object, Object propertyValue, ValidationResultFactory resultFactory) {
    if ((propertyValue != null) && !propertyValue.toString().isEmpty()) {
      String email = String.valueOf(propertyValue);
      if (!email.contains("@")) {
        resultFactory.addResult("InvalidEmail");
      }
    }
  }
}
